File: bar_frontend/src/app/model/Ingredient.ts
export class Ingredient {
  ingredientId: number;
  ingredientName: string;
  ingredientAlcohol: number;
  unitOfMeasurement: string;


  constructor(ingredientName: string, ingredientAlcohol: number, unitOfMeasurement: string, ingredientId?: number) {
    this.ingredientId = ingredientId;
    this.ingredientName = ingredientName;
    this.ingredientAlcohol = ingredientAlcohol;
    this.unitOfMeasurement = unitOfMeasurement;
  }

  // toJSON(): any {
  //   const { ingredientId, ...rest } = this;
  //   return {
  //     '@class': 'com.gbyzzz.bar_web_app.bar_backend.dto.IngredientDTO',
  //     ...rest
  //   };
  // }
}

File: bar_spring/src/main/java/com/gbyzzz/bar_spring/entity/Ingredient.java
package com.gbyzzz.bar_spring.entity;

import lombok.*;

import javax.persistence.*;

@Entity
@Table(name = "ingredients")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Ingredient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ingredient_id")
    private Long ingredientId;

    @Basic
    @Column(name = "ingredient_name")
    private String ingredientName;

    @Basic
    @Column(name = "ingredient_alcohol_percentage")
    private int ingredientAlcohol;

    @Basic
    @Column(name = "unit_of_measurement")
    private String unitOfMeasurement;
}
